### PR TITLE
Add support for non-github repos

### PR DESCRIPTION
**Pull request by YoussB**
- With this [PR](https://github.com/concourse/resource-types/pull/47), I noticed that the current code wouldn't work with non-github repositories and even panicking the whole server.

- This is because we were trying to get the stars of a repo regardless where it is hosted.

- This PR will allow to only call the Github graphql when the repo is hosted under Github.
- Also, for the aforementioned PR, the URL doesn't contain a part for the owner, I also handled that.

- @jomsie, we might need to do some tweaks to the UI, as when I return the `owner` as empty string it kinda messes with the way it is displayed. I also noticed that there is no way to hide the Github stars icon, which in this case will be irrelevant.

### CODE REVIEW

**Comment by jomsie** [comment]
Hmmm, I was dreading this day would come. I haven't looked at the elm code in forever, maybe I'll take a gander this afternoon and see what's what. 

UI will definitely need some changes....

**Comment by jomsie** [comment]
Quick question!

We can do this one of two ways, either detect what hosting service it uses on the backend, and the frontend can continue to be _stupid_ (just look at resource type host for displaying stuff), or we can do it on the frontend, then we'd have some logic (and subsequently a whole new world of testing). 

What do you think?

**Comment by ivan-avalos** [comment]
What about adding a `owner` option to the YAMLs as a fallback?

**Comment by YoussB** [comment]
@jomsie, 
- if we're talking about the owner, then the current will work with github, gitlab, and bitbucket.
  - I was thinking about adding the owner to the yaml as well, thx @ivan-avalos 
- if this is for the sake of stars, I think we can just skip the stars in case of non-github.

**Comment by jomsie** [comment]
@YoussB I'm cool with anything, but I just want to know where to put the logic whether or not its github.

On the front end it'd be adding some basic "does the resource type url start with https://github.com" for github, and if so, don't show stars / icon.

But we don't really have any logic like that on the front end yet iirc. Just wondering if we want to add a new field from the backend telling the front end what host service the RT is on.

**Comment by YoussB** [comment]
I can add a hosting field, shouldn't be that big of a deal.

**Comment by jomsie** [comment]
@YoussB I'm going to work on this (this pr review + ui) on freedom friday!

File: warehouse/githubwrapper/wrapper.go [hunk 1]
```diff
@@ -47,14 +47,16 @@ func (w wrapper) GetStars(repoStarsMap map[string]int) (map[string]int, error) {
...
-			Tag:  reflect.StructTag(fmt.Sprintf(`graphql:"i%d: repository(owner: \"%s\", name: \"%s\")"`, i, strs[0], strs[1])),
-		})
-		i++
+	for k, isGithub := range repoStarsMap {
+		if isGithub {
+			strs := strings.Split(k, "/")
+			arr = append(arr, reflect.StructField{
+				Name: fmt.Sprintf("I%d", i),
+				Type: ghReturnReflection,
+				Tag:  reflect.StructTag(fmt.Sprintf(`graphql:"i%d: repository(owner: \"%s\", name: \"%s\")"`, i, strs[0], strs[1])),
```
[the remark this answers is not in the document]
Comment (YoussB) [reply warehouse/githubwrapper/wrapper.go:60]: I added a little documentation to explain as much as I recall of what we had to do. And no, the function is already written in a weird way, it is not you 😁 .

**Pull request merged**